src/config_libconfig.c

```diff
@@ -37,6 +36,48 @@ static inline int lcfg_lookup_bool(const config_t *config, const char *path, boo
 	return ret;
 }
 
+const char *xdgConfigHome(void) {
+	char *xdgh = getenv("XDG_CONFIG_HOME");
+	char *home = getenv("HOME");
+
+	if (!xdgh) {
+		if (!home) return NULL;
+
+		xdgh = malloc(strlen(home) + 8 + 1);
+
+		if (!xdgh) return NULL;
```

Comment:
You can use `cvalloc` from `utils.h` here, it aborts when allocation fails.

```suggestion
		xdgh = cvalloc(strlen(home) + 8 + 1);
```